src/category_theory/adjunction/basic.lean

```diff
@@ -204,16 +204,73 @@ def id : 𝟭 C ⊣ 𝟭 C :=
   unit := 𝟙 _,
   counit := 𝟙 _ }
 
+/-- If F and G are naturally isomorphic functors, establish an equivalence of hom-sets. -/
+@[reducible]
```

Comment:
I'd prefer installing the appropriate `simp` lemmas for the projections, over marking these definitions as `@[reducible]`. It's slightly more work to set up, but avoids the possibility of a tactic accidentally unfolding to produce the entire structure.

(Basically, `@[reducible]` is a code smell... It probably warrants an explanation of why it's unavoidable, if it really is.)